### PR TITLE
Ability to control if separate threads should be created to handle events

### PR DESCRIPTION
**Pull request by serhiip**
I was using this library for a while now and in my case I needed to synchronise all of the threads (MVar)created by hfnotify library. But why create threads at all if I know I'll execute them synchronously. This PR allows to control that via new `WatchConfig` parameter

### CODE REVIEW

File: src/System/FSNotify.hs [hunk 1]
```diff
@@ -207,8 +208,8 @@ threadChan listenFn (WatchManager db listener cleanupVar) path actPred action =
         , stopListener >> cleanThisUp
         )
 
-readEvents :: EventChannel -> Action -> IO ()
-readEvents chan action = forever $ do
+readEvents :: WatchConfig -> EventChannel -> Action -> IO ()
+readEvents (WatchConfig _ _ _ True) chan action = forever $ do
```
Comment (thomasjm) [inline src/System/FSNotify.hs]: Could you change this to be `(WatchConfig {confThreadPerEvent=True})` and the same for `False`?

File: src/System/FSNotify/Types.hs [hunk 2]
```diff
@@ -69,6 +69,9 @@ data WatchConfig = WatchConfig
   , confUsePolling :: Bool
     -- ^ Force use of polling, even if a more effective method may be
     -- available. This is mostly for testing purposes.
+  , confThreadPerEvent :: Bool
+    -- ^ If separate thread should be created for each event
```
Comment (thomasjm) [inline src/System/FSNotify/Types.hs]: Nit: reword a little. How about "Fork a separate thread for each event when executing the user supplied event handler"

File: src/System/FSNotify.hs (hunk 1 above)
Comment (serhiip) [reply src/System/FSNotify.hs]: `hlint` also suggested to remove the `()` around record pattern. So I did that also

File: src/System/FSNotify/Types.hs (hunk 2 above)
Comment (serhiip) [reply src/System/FSNotify/Types.hs]: Yes, that sounds better

**Comment by serhiip** [comment]
I updated the PR with recommended changes. Should I also squash my commits into one?

**Comment by thomasjm** [comment]
No squash needed, I'll merge this now. I'll try to cut a release soon--I might fold in another change or two while we're making breaking API changes.

**Pull request merged**